### PR TITLE
[#1837564] Migrate app service related components to app service library

### PR DESCRIPTION
**Pull request by Flanker32**
- Migrate app service properties view to app service library
- Migrate validation utils to track2 SDK
- Migrate app service docker view to app service library
- Remove SDK pricing in web app configuration

[AB#1837564](https://dev.azure.com/mseng/a4d27ce2-a42d-4b71-8eef-78cee9a9728e/_workitems/edit/1837564)

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/function/runner/deploy/FunctionDeployConfiguration.java [hunk 1]
```diff
@@ -149,7 +151,7 @@ public void setFunctionId(String id) {
     public void readExternal(Element element) throws InvalidDataException {
         this.functionDeployModel = Optional.ofNullable(element.getChild("FunctionDeployModel"))
                 .map(e -> XmlSerializer.deserialize(e, FunctionDeployModel.class))
-                .orElseGet(() -> Optional.ofNullable(element)
+                .orElseGet(() -> Optional.of(element)
```
Comment (shenqianjin) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/function/runner/deploy/FunctionDeployConfiguration.java:154]: why change this line ? 
it's exactly sure that element is never null?

Comment (Flanker32) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/function/runner/deploy/FunctionDeployConfiguration.java:154]: Yes, defined in RunConfigurationBase interface, missed in our abstract parent class `public void writeExternal(@NotNull Element element) {`

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/appservice/AppServiceComboBox.java [hunk 2]
```diff
@@ -72,12 +73,7 @@ protected String getItemText(final Object item) {
 
     protected abstract void createResource();
 
-    public class AppCombineBoxRender extends SimpleListCellRenderer {
-        private final JComboBox comboBox;
-
-        public AppCombineBoxRender(JComboBox comboBox) {
-            this.comboBox = comboBox;
-        }
+    public static class AppCombineBoxRender extends SimpleListCellRenderer {
```
Comment (Flanker32) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/appservice/AppServiceComboBox.java:76]: typo

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/lib/webapp/WebAppService.java [hunk 3]
```diff
@@ -85,4 +91,13 @@ public static WebAppSettingModel convertConfig2Settings(final WebAppConfig confi
         settings.setTargetPath(config.getApplication() == null ? null : config.getApplication().toString());
         return settings;
     }
+
+    public String getRuntimeDisplayName(@Nonnull final Runtime runtime) {
+        final String os = runtime.getOperatingSystem().getValue();
+        final String javaVersion = runtime.getJavaVersion() == JavaVersion.OFF ? null : String.format("Java %s", runtime.getJavaVersion().getValue());
```
Comment (Flanker32) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/lib/webapp/WebAppService.java:97]: check to equals

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/util/ValidationUtils.java [hunk 4]
```diff
@@ -70,9 +73,9 @@ public static void validateAppServiceName(String subscriptionId, String appServi
         if (!isValidAppServiceName(appServiceName)) {
             cacheAndThrow(appServiceNameValidationCache, cacheKey, message("appService.subscription.validate.invalidName"));
         }
-        final Azure azure = AuthMethodManager.getInstance().getAzureManager().getAzure(subscriptionId);
-        final ResourceNameAvailabilityInner result = azure.appServices().inner()
-            .checkNameAvailability(appServiceName, CheckNameResourceTypes.MICROSOFT_WEBSITES);
+        final AzureResourceManager azureResourceManager = Azure.az(AzureAppService.class).getAzureResourceManager(subscriptionId);
```
Comment (Flanker32) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/util/ValidationUtils.java:76]: Move to lib

**Review by shenqianjin — APPROVED** [review]

**Pull request merged**